src/main/java/mServer/tool/M3U8Utils.java

```diff
@@ -0,0 +1,92 @@
+package mServer.tool;
+
+import mServer.crawler.sender.newsearch.Qualities;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A class with some Utils to work with M3U8 urls.
+ */
+public class M3U8Utils {
+	public static final String M3U8_WDR_URL_BEGIN = "http://adaptiv.wdr.de/i/medp/";
+	private static final String REGEX_FIRST_USELESS_COMMA = "^,";
+	private static final String M3U8_WDR_QUALITIES_USELESS_END = ",.mp4.csmil";
+	public static final String WDR_MP4_URL_PATTERN = "http://ondemand-%s.wdr.de/medp/%s/%s/%s/%s.mp4";
+
+	private M3U8Utils() {
+		super();
+	}
+
+	/**
+	 * If the URL follows the following structure it will be used to generate
+	 * MP4 URLS from it.<br>
+	 * This structure is needed:
+	 * <code>http://adaptiv.wdr.de/i/medp/[region]/[fsk]/[unkownNumber]/[videoId]/,[Qualitie 01],[Qualitie 02],[Qualitie ...],.mp4.csmil/master.m3u8</code><br>
+	 *
+	 * @param aWDRM3U8Url
+	 *            The M3U8 URL.
+	 * @return A Map containing the URLs and Qualities which was found. An empty
+	 *         Map if nothing was found.
+	 */
+	public static Map<Qualities, String> gatherUrlsFromWdrM3U8(String aWDRM3U8Url) {
+		Map<Qualities, String> urlAndQualities = new EnumMap<>(Qualities.class);
+		if (aWDRM3U8Url.startsWith(M3U8_WDR_URL_BEGIN)) {
+			String m3u8Url = aWDRM3U8Url.replace(M3U8_WDR_URL_BEGIN, "");
+			urlAndQualities.putAll(convertM3U8Url(m3u8Url));
+		}
+		return urlAndQualities;
+	}
+
+	private static Map<? extends Qualities, ? extends String> convertM3U8Url(String m3u8Url) {
+		Map<Qualities, String> urlAndQualities = new EnumMap<>(Qualities.class);
+		String[] splittedM3U8Url = StringUtils.split(m3u8Url, '/');
+		if (splittedM3U8Url.length >= 6) {
+			String region = splittedM3U8Url[0];
+			String fsk = splittedM3U8Url[1];
+			String unkownNumber = splittedM3U8Url[2];
+			String videoId = splittedM3U8Url[3];
+
+			// From lowest to best quality;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)